### PR TITLE
RK-6363 - added docs about auto-fetching

### PR DESCRIPTION
**Pull request by yonitwis**

### CODE REVIEW

File: docs/source-repos.md [hunk 1]
```diff
@@ -4,9 +4,40 @@ title: Source Repositories
...
+Rookout allows you to easily load sources from your local file system or your git provider. 
+When you integrate your source code into Rookout, it remains between your code repository and your local browser.
+**Rookout never sees nor has access to your source code. In fact, your code never even reaches the Rookout servers.**
+ 
 
-Rookout allows you to easily load sources from your local file system or your git provider.  This is performed directly in our Web based IDE.
+## Source Code Fetching 
+### Git Providers
+
+Rookout supports the following Git Providers:
```
Comment (Rookeret73) [inline docs/source-repos.md:17]: Rookout integrates directly to the cloud editions of the following Git Providers:
1. Github
2. Gitlab
3. Bitbucket

In addition, Rookout offers a desktop app for fetching source repositories from a local file system.
This allows fetching source code from local editions of Git providers, as well as from Perforce.

File: docs/source-repos.md [hunk 2]
```diff
@@ -4,9 +4,40 @@ title: Source Repositories
...
+1. Github
+2. BitBucket
+3. GitLab
+
+##### Automatic Fetching
+To configure automatic fetching sources please add the following environment variables while deploying Rookout:
```
Comment (Rookeret73) [inline docs/source-repos.md:23]: To automatically fetch source code repositories from cloud based Github, Bitbucket or Gitlab, use the following environment variables while deploying rookout:

File: docs/source-repos.md [hunk 3]
```diff
@@ -4,9 +4,40 @@ title: Source Repositories
...
+1. ```ROOKOUT_COMMIT```  - String that indicates your git commit
+2. ```ROOKOUT_REMOTE_ORIGIN``` - String that indicates your git remote origin
+
+
+After adding those environment variables, the sources will be loaded automatically once the instance will be filtered.
+
+
+##### Manual Fetching
+You can load your sources manually in the web IDE as explained in the following video.
+In that case, you need to make sure that you have load the sources that match the sources of the running instance.
```
Comment (Rookeret73) [inline docs/source-repos.md:33]: When fetching source repositories manually, make sure to fetch the version of the code that is deployed in the app you are trying to debug.

File: docs/source-repos.md [hunk 4]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
 sidebar_label: Source Repositories
 ---
 
-Each Project contains the source code you wish to debug.
+## Introduction
+Each Debug Session contains the source code you wish to debug.
```
Comment (LiranLast) [inline docs/source-repos.md:8]: Feels too focused on our terminology instead of the user :)

File: docs/source-repos.md [hunk 5]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
...
+Rookout allows you to easily load sources from your local file system or your git provider. 
+When you integrate your source code into Rookout, it remains between your code repository and your local browser.
+**Rookout never sees nor has access to your source code. In fact, your code never even reaches the Rookout servers.**
+ 
 
-Rookout allows you to easily load sources from your local file system or your git provider.  This is performed directly in our Web based IDE.
+## Source Code Fetching 
+### Git Providers
```
Comment (LiranLast) [inline docs/source-repos.md:15]: Let's focus on the UX and instructions upfront and leave the supported tech section for the end?

File: docs/source-repos.md [hunk 6]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
...
+
+Rookout integrates directly to the cloud editions of the following Git Providers:
+1. Github
+2. BitBucket
+3. GitLab
+
+In addition, Rookout offers a desktop app for fetching source repositories from a local file system.
```
Comment (LiranLast) [inline docs/source-repos.md:22]: Let's create a unified narrative around non-cloud hosted providers?

File: docs/source-repos.md [hunk 7]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
...
+This allows fetching source code from local editions of Git providers, as well as from Perforce.
+
+##### Automatic Fetching
+To automatically fetch source code repositories from cloud based Github, Bitbucket or Gitlab, use the following environment variables while deploying rookout:
```
Comment (LiranLast) [inline docs/source-repos.md:26]: Doesn't this apply always? For on-prem as well as Perforce?

File: docs/source-repos.md [hunk 8]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
...
+1. ```ROOKOUT_COMMIT```  - String that indicates your git commit
+2. ```ROOKOUT_REMOTE_ORIGIN``` - String that indicates your git remote origin
+
+
+After adding those environment variables, the sources will be loaded automatically once the instance will be filtered.
```
Comment (LiranLast) [inline docs/source-repos.md:31]: Can't the user do it using APIs as well? Why not have snippets like labels?

File: docs/source-repos.md [hunk 9]
```diff
@@ -4,9 +4,42 @@ title: Source Repositories
...
+
+Rookout integrates directly to the cloud editions of the following Git Providers:
+1. Github
+2. BitBucket
+3. GitLab
+
+In addition, Rookout offers a desktop app for fetching source repositories from a local file system.
+This allows fetching source code from local editions of Git providers, as well as from Perforce.
+
+##### Automatic Fetching
```
Comment (LiranLast) [inline docs/source-repos.md:25]: Reference for building into the JARs is missing

File: docs/source-repos.md [hunk 10]
```diff
@@ -4,9 +4,57 @@ title: Source Repositories
...
+After adding those environment variables, the sources will be loaded automatically once the instance will be filtered.
+
+
+##### Manual Fetching
+You can load your sources manually in the web IDE as explained in the following video.
+When fetching source repositories manually, make sure to fetch the version of the code that is deployed in the app you are trying to debug.
+<iframe width="600" height="300" src="https://www.youtube.com/embed/divAqo048eA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+### Local FileSystem - Rookout Desktop App 
+If you are using a git provider that Rookout does not support, you can associate source code files from the local filesystem into a Rookout debug session.
```
Comment (LiranLast) [inline docs/source-repos.md:40]: There are no more of those :)

File: docs/source-repos.md [hunk 11]
```diff
@@ -4,9 +4,57 @@ title: Source Repositories
...
+To do that, please download and install Rookout Desktop App. See the following video to learn how to download Rookout Desktop App:
+<iframe width="600" height="300" src="https://www.youtube.com/embed/mkMpzQPNcsI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Packaging Sources
```
Comment (LiranLast) [inline docs/source-repos.md:44]: .Net?

**Comment by LiranLast** [comment]
@yonitwis you need to open a ticket to merge

**Pull request merged**